hub-detect/src/main/groovy/com/blackducksoftware/integration/hub/detect/util/DetectFileFinder.java

```diff
@@ -92,33 +69,27 @@ public boolean containsAllFilesToDepth(final String sourcePath, final int maxDep
             return false;
         }
 
-        boolean containsFiles = true;
-        for (final String filenamePattern : filenamePatterns) {
-            final List<File> foundFiles = findFilesToDepth(sourceDirectory, filenamePattern, maxDepth);
-            if (null == foundFiles || foundFiles.isEmpty()) {
-                containsFiles = false;
-                logger.debug(String.format("No file detected: %s in %s", filenamePattern, sourcePath));
-                break;
-            }
-        }
-        return containsFiles;
+        return Arrays.stream(filenamePatterns).allMatch(pattern -> !findFilesToDepth(sourceDirectory, pattern, maxDepth).isEmpty());
     }
 
-    public File findFile(final String sourcePath, final String filenamePattern) {
-        final File sourceDirectory = new File(sourcePath);
-        return findFile(sourceDirectory, filenamePattern);
+    public File findFile(final String sourceDirectory, final String filenamePattern) {
+        return findFile(new File(sourceDirectory), filenamePattern);
     }
 
     public File findFile(final File sourceDirectory, final String filenamePattern) {
+        final File foundFile;
         final List<File> foundFiles = findFiles(sourceDirectory, filenamePattern);
         if (foundFiles == null || foundFiles.isEmpty()) {
             logger.debug(String.format("Could not find any matches for %s in %s", filenamePattern, sourceDirectory.getAbsolutePath()));
-            return null;
-        } else if (foundFiles.size() > 1) {
-            logger.debug(String.format("Found multiple matches for %s in %s", filenamePattern, sourceDirectory.getAbsolutePath()));
-            logger.debug(String.format("Using %s", foundFiles.get(0)));
+            foundFile = null;
+        } else {
+            foundFile = foundFiles.get(0);
+            if (foundFiles.size() > 1) {
+                logger.debug(String.format("Found multiple matches for %s in %s", filenamePattern, sourceDirectory.getAbsolutePath()));
+                logger.debug(String.format("Using %s", foundFile));
```

Comment:
Are we sure we want to return a file if we found multiple matches? Seems potentially unpredictable.

Reply:
I realize the old one did the same thing so it's probably fine.